contracts/credit/CreditFacadeV3.sol

```diff
@@ -729,17 +723,20 @@ contract CreditFacadeV3 is ICreditFacadeV3, ACLNonReentrantTrait {
     }
 
     /// @dev `ICreditFacadeV3Multicall.withdrawCollateral` implementation
-    function _withdrawCollateral(address creditAccount, address token, uint256 amount, address to) internal {
+    function _withdrawCollateral(address creditAccount, bytes calldata callData) internal {
+        (address token, uint256 amount, address to) = abi.decode(callData, (address, uint256, address)); // U:[FA-36]
+
         if (amount == type(uint256).max) {
             amount = IERC20(token).safeBalanceOf(creditAccount);
-            if (amount <= 1) return;
-            unchecked {
-                --amount;
+            if (amount >= 1) {
```

Comment:
Wouldn't that cause a revert if the balance of the credit account is `1`? What about previous gas optimisations used here?

Reply:
I guess the point is to avoid no-op withdrawals when the balance is `<= 1` (one can still specify exactly `amount == 1` if this 1 wei is of any value).

Reply:
That's the main reason why I do not like to have custom logics like this one.

As @lekhovitsky said, people can just specify the **exact** number of tokens to withdraw to avoid this revert, and definitely it's an edge case (given that you have to have just `1 wei` in the balance of your CA for this token) but still...

@lekhovitsky with gas as low as `0.805 gwei` like we have right now, is it really worth having this custom behavior?

Reply:
@StErMi Still don't see a single an issue with it, tbh